### DISCUSSION TITLE
The AudioTools basic-a2dp-i2s example crashes

### DISCUSSION THREAD

**Discussion by vorosj** [pschatzmann/arduino-audio-tools#1022] [Q&A] [answered]
Hi, I am playing with your audio library, great fun, thank you. Most examples are working very well, but I found an example that keeps crashing-restart (core1panic, dividebyzero error): arduino-audio-tools/examples/examples-basic-api/basic-a2dp-i2s
The similar basic-a2dp-mixer-i2sexample crashes too, but with softreset, so I am afraid this error message is not really informative.

The ESP32-A2DP/examples/bt_music_receiver_simple is properly working, so the hw and probably the compiler settings are OK.
(different I2S pinout, I use the i2s_pin_config_t and the 14,15,22 pinout) 
My hw:
ESP - 32S board, HiLetgo PCM5102 I2S IIS Lossless Digital Audio DAC Decoder
PlatformIO IDE, the .ini:
[env:nodemcu-32s]
platform = espressif32
board = nodemcu-32s
framework = arduino
;board_build.partitions = huge_app.csv
monitor_speed = 115200 ;922190
lib_deps = https://github.com/pschatzmann/arduino-audio-tools
            https://github.com/pschatzmann/ESP32-A2DP
(in some examples I set higher uart speed, because the serial messages resulted in stutter)

Any idea, what is the problem?
regards, 
   Jozsef

**Comment by pschatzmann** [comment, marked as answer]
I would recommend that you uncomment the partition setting: A2DP usually does not fit into the default setting any more!
If this does not help maybe try another board.

The correct way to resolve the issues with the playback is to set the log level to Warning.

**Reply by vorosj** [reply]
Thank you for the fast reply. Of course I tried that (thats why there is the line), and tried another 32S board, still crashes. I'll try a different type, maybe an ESP32 S2 if I find one.
Also I'll check the Arduino IDE, maybe PlatformIO has different default settings or something.

_The correct way to resolve the issues with the playback is to set the log level to Warning._
Sure, but I was interested in the values it's writing... :-)

**Comment by pschatzmann** [comment]
I am not really familiar with these boards. So far I have tested some VROOM, WROVER and A1S based boards.
The C3, S2 and S3 processors do not support any A2DP as indicated in the [documentation](https://github.com/pschatzmann/ESP32-A2DP/wiki/Support-for-esp32-c3%3F)

**Comment by pschatzmann** [comment]
This could also be related to [this](https://github.com/pschatzmann/ESP32-A2DP/issues/494) - though it's strange that you did not get any linker error using platformio
